scripts/make/build_c.mk

```diff
@@ -65,12 +67,16 @@ app-objs := main.o
 
 app-objs := $(addprefix $(APP)/,$(app-objs))
 
+$(ulib_hdr): _cargo_build
+
 $(APP)/%.o: $(APP)/%.c $(ulib_hdr)
 	$(call run_cmd,$(CC),$(CFLAGS) $(APP_CFLAGS) -c -o $@ $<)
 
+$(rust_lib): _cargo_build
+
 $(OUT_ELF): $(c_lib) $(rust_lib) $(libgcc) $(app-objs)
 	@printf "    $(CYAN_C)Linking$(END_C) $(OUT_ELF)\n"
-	$(call run_cmd,$(LD),$(LDFLAGS) $^ -o $@)
+	$(call run_cmd,$(LD),$(LDFLAGS) $(c_lib) $(rust_lib) $(libgcc) $(app-objs) -o $@)
```

Comment:
Is this different from `^`?

Reply:
`$(rust_lib)` depends on `_cargo_build`, which is a .PHONY target. `$^` does not take this into consideration, and will cause `_cargo_build` to appear in the final substitution, leading to build failure.